### PR TITLE
feature: Remove pending transaction contention at commit

### PR DESCRIPTION
**Pull request by dhedey**
Follow up to https://github.com/radixdlt/babylon-gateway/pull/484

Fixes:
* The issue where a commit was contentious with other pending transaction updates, which could block commit
* That rejected transactions weren't storing their rejection messages on Pending Transactions
* Also fixes random other bug I was seeing where requests for 0 transactions were sent to the Core API

In general, after all these simplifications, pending transactions should be a lot more performant now.

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.DataAggregator/Workers/NodeWorkers/NodeTransactionFetchWorker.cs [hunk 1]
```diff
@@ -175,6 +175,15 @@ private async Task FetchTransactions(CancellationToken cancellationToken)
             fetchMaxBatchSize
         );
 
+        if (batchSize == 0)
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.DataAggregator/Workers/NodeWorkers/NodeTransactionFetchWorker.cs:178]: This fixes a misc bug I noted where the Aggregator was requesting 0 transactions from the Core API

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 2]
```diff
@@ -1756,7 +1756,20 @@ components:
...
-            An error message received for a rejection or failure during transaction execution.
+            The initial error message received for a rejection or failure during transaction execution.
+            This will typically be the useful error message, explaining the root cause of the issue.
+            Please note that presence of an error message doesn't imply that this payload
+            will definitely reject or fail. This could represent an error during a temporary
+            rejection (such as out of fees) which then gets resolved (e.g. by depositing money
+            to pay the fee), allowing the transaction to be committed.
+          type: string
+          nullable: true
+        latest_error_message:
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1767]: Could you please update CHANGELOG.md file with some brief summarization of pending TX changes?

Comment (dhedey) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1767]: Yes, great shout.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs [hunk 3]
```diff
@@ -158,90 +158,65 @@ public async Task<CommitTransactionsReport> CommitTransactions(ConsistentLedgerE
...
             .Where(pt => payloadHashes.Contains(pt.PayloadHash))
+            .Where(pt => pt.LedgerDetails.PayloadLedgerStatus == PendingTransactionPayloadLedgerStatus.PermanentlyRejected)
+            .Select(pt => new
+            {
+                pt.PayloadHash,
+                pt.LedgerDetails.PayloadLedgerStatus,
+                pt.GatewayHandling.FirstSubmittedToGatewayTimestamp,
+                pt.LedgerDetails.LatestRejectionTimestamp,
+                pt.LedgerDetails.LatestRejectionReason,
+            })
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:180]: As we update TXs with completely separate UPDATE query guess we can add `.NoTracking()`?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs [hunk 4]
```diff
@@ -158,90 +158,65 @@ public async Task<CommitTransactionsReport> CommitTransactions(ConsistentLedgerE
...
+                $@"
+UPDATE pending_transactions
+    SET
+        payload_status = 'committed',
+        intent_status = 'committed',
+        commit_timestamp = {timestamp},
+        resubmit_from_timestamp = NULL,
+        handling_status_reason = 'Concluded as committed'
+    WHERE
+        payload_hash IN (SELECT UNNEST({payloadHashes}))
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:213]: `payload_hash = ANY(UNNEST(...))`?

Comment (dhedey) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:213]: Sure, if that's faster :)

**Review by krzlabrdx — APPROVED** [review]

**Pull request merged**